README.md

```diff
@@ -24,47 +24,105 @@ TODO
+```js
+import { createStore } from 'k-simple-state'
 
-const saveToLocalStorage = store => next => (action) => {
-  // dispatch action
-  const res = next(action)
+// create a store of todos
+const store = createStore({
+  todos: { type: 'keyValue', key: 'id' },
+})
 
-  // save store to localStorage
-  window.localStorage.setItem('redux-store', JSON.stringify(store.getState()))
+// this next line will dispatch a k-redux-factory action and update the store
+store.todos.add({ id: 2, label: 'write a better README' })
 
-  // return action for next middlewares
-  return res
-}
+// you can retrieve data like that
+const todo = store.todos.get(2)
+```
 
-const state = createState({
-  data: {
-    todos: { type: 'simpleObject' },
-    context: { type: 'keyValue', key: 'key' },
-  },
-  ui: {
-    bla: { type: 'simpleObject' },
-    testCustomReducer: (state, action) => {
-      if (!state) return 'oui'
-      if (action.type === 'testCustom') return action.payload
-      return state
-    },
-  },
-  config: state => state || {},
-}, {
-  hideRedux: true,
-  middlewares: [saveToLocalStorage],
-  init: {
-    config: { this: 'is', from: { init: 'state' } },
-  },
+### Connect it with ReactJS
+1. Create a store
+
+**store.js**
+```js
+import { createStore, keyValue } from 'k-simple-state'
+
+export default createStore({
+  todos: keyValue({ key: 'id' }),
 })
+```
+
+2. Provide the store to React context
+
+**app.jsx**
+```js
+import { provider } from 'k-simple-state/react'
+import store from './store'
+import TodosContainer from './todos.container'
+
+const App = () => <TodosContainer />
+
+// use `provider` HoC to inject store to the React context
+export default provider(store)(App)
+```
+
+3. Use `inject` to interact with the store, wrap your `<Todos />` graphical component in a container
+
+**todos.container.js**
+```js
+import { inject } from 'k-simple-state/react'
+import Todos from './todos'
+
+// `inject` is an HoC, like `connect` from react-redux,
+// it takes one parameter and returns props to be injected to the wrapped component
+// (FYI: props injected to the wrapped component are added to the props given by the parent)
+export default inject(store => ({
+  todos: store.todos.getAsArray(),
+  onAdd: event => store.todos.add({ id: Date.now(), label: event.target.value }),
+}))(Todos)
+```
+
+4. Write your classical React JSX component (here, as pure function)
+
+**todos.jsx**
+```js
+import React from 'react'
+import PropTypes from 'prop-types'
+
+const Todos = ({ todos, onAdd }) => (
+  <div>
+    <input onBlur={onAdd} />
+    <ul>
+      {todos.map(todo => <li>{todo.label}</li>})}
+    </ul>
+  </div>
+)
+
+Todos.propTypes = {
+  todos: PropTypes.array.isRequired,
+  onAdd: PropTypes.func.isRequired,
+}
 
-state.ui.bla.set({ one: 'object', so: 'cool', right: true })
-state.dispatch(state.ui.bla.set({ this: 'is', with: 'redux' }))
-state.dispatch({ type: 'testCustom', payload: 'ok ça marche' })
+export default Todos
 ```
```

Comment:
You may should add a third example with a raw redux reducer.
And maybe, it should be the first one, because, everybody reading the doc will easly userstand the first example.
And in the second one, you could show how it is simpler with k-redux-factory helpers

Reply:
Yeah I can add a lot of new example, but later :)
 - Raw reducers (I don't agree right now that it should be the first one, unless we create this README.md like a tutorial where each example is a better usage (or more sophisticated usage)
 - listeners
 - etc

Reply:
I think that add a `simpleObject` example can be more useful than a raw reducer example :-)